src/Common/src/Interop/Unix/Interop.IOErrors.cs

```diff
@@ -8,7 +8,7 @@
 
 internal static partial class Interop
 {
-    private static bool CheckIo(ErrorInfo errorInfo, string path, bool isDirectory, Func<ErrorInfo, ErrorInfo> errorRewriter)
+    private static void CheckIo(ErrorInfo errorInfo, string path, bool isDirectory, Func<ErrorInfo, ErrorInfo> errorRewriter)
```

Comment:
We should rename this.  It's no longer about checking and more about throwing.  Seems like it should really just be an overload of GetExceptionForIoErrno.